.danger.php

```diff
@@ -43,78 +41,18 @@
-
-                if ($context->platform->raw['squash'] === true && in_array('github', $labels, true)) {
-                    $context->failure('GitHub PRs are not allowed to be squashed');
-                }
-            },
-        ]
-    ))
-    /**
-     * MRs that target a release branch that is not trunk should have a thread with link to a trunk MR
-     * to disable this rule you can add the no-trunk label
-     */
-    ->useRule(new Condition(
-        function (Context $context) {
-            $labels = array_map('strtolower', $context->platform->pullRequest->labels);
-
-            return $context->platform instanceof Gitlab
-                && !\in_array('no-trunk', $labels, true)
-                && preg_match('#^6\.\d+.*|saas/\d{4}/\d{1,2}$#', $context->platform->raw['target_branch']);
-        },
-        [
-            function (Context $context): void {
-                if (str_contains($context->platform->pullRequest->body, '/shopware/6/product/platform/-/merge_requests/')) {
-                    return;
-                }
-
-                $found = false;
 
-                foreach ($context->platform->pullRequest->getComments() as $comment) {
-                    if (str_contains($comment->body, '/shopware/6/product/platform/-/merge_requests/')) {
-                        $found = true;
-                        break;
-                    }
-                }
-
-                if (!$found) {
-                    $context->failure('This MR should have a dependency on a trunk MR. Please add a thread with a link');
-                }
-            },
-        ]
-    ))
     ->useRule(new Condition(
         function (Context $context) {
             $labels = array_map('strtolower', $context->platform->pullRequest->labels);
 
-            return $context->platform instanceof Gitlab && !\in_array('github', $labels, true);
+            return !\in_array('skip-danger-phpstan-baseline', $labels, true);
         },
         [
             function (Context $context): void {
-                $files = $context->platform->pullRequest->getFiles();
-
-                /** @var Gitlab $gitlab */
-                $gitlab = $context->platform;
-
-                $phpstanBaseline = new GitlabFile(
-                    $gitlab->client,
-                    $_SERVER['CI_PROJECT_ID'],
-                    'phpstan-baseline.neon',
-                    $gitlab->raw['sha']
-                );
-
-                $fileNames = $files->map(fn (File $f) => $f->name);
-
                 $filesWithIgnoredErrors = [];
-                foreach ($fileNames as $fileName) {
-                    if (str_contains($phpstanBaseline->getContent(), 'path: ' . $fileName)) {
+                foreach ($context->platform->pullRequest->getFiles()->map(fn (File $f) => $f->name) as $fileName) {
+                    if (str_contains($context->platform->pullRequest->getFile('phpstan-baseline.neon')->getContent(), 'path: ' . $fileName)) {
```

Comment:
Is that the same as before? To me that looks like it would just get the baseline from the pr, but in most cases it won't be in there?

Reply:
i would store the getFile once outside of the loop. otherwise it downloads on every loop that file

Reply:
done